protocol/transport/filters_manager.go

```diff
@@ -230,6 +231,33 @@ func (s *FiltersManager) Remove(filters ...*Filter) error {
 	return nil
 }
 
+// Remove remove all the filters associated with a chat/identity
+func (s *FiltersManager) RemoveNoListenFilters() error {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	var filterIDs []string
+	var filters []*Filter
+
+	for _, f := range filters {
+		if !f.Listen {
+			filterIDs = append(filterIDs, f.FilterID)
+			filters = append(filters, f)
+		}
+	}
+	if err := s.service.UnsubscribeMany(filterIDs); err != nil {
+		return err
+	}
+
+	for _, filter := range filters {
+		if filter.SymKeyID != "" {
+			s.service.DeleteSymKey(filter.SymKeyID)
+		}
+		delete(s.filters, filter.ChatID)
+	}
```

Comment:
This uses 3 for loops to go through the same information. Once to find if we don't listen, once to unsubscribe and once to delete the filters. Would the below do the same with less loops?

```suggestion
	for _, f := range filters {
		if !f.Listen {
			filterIDs = append(filterIDs, f.FilterID)
			filters = append(filters, f)
			
			if err := s.service.Unsubscribe(f.FilterID); err != nil {
				return err
			}
			
			if filter.SymKeyID != "" {
				s.service.DeleteSymKey(filter.SymKeyID)
			}
			delete(s.filters, filter.ChatID)
		}
	}
```

Reply:
Not quite, because `Unsubscribe` trigger a waku message, so that would be `n` messages, so it's best to collect and unsubscribe all in one go, producing a single waku message, and after, we want to delete filters only if we managed to unsubscbribe, otherwise we might be in some inconsistent state, so not sure we can avoid the looping